store/rootmulti/store_test.go

```diff
@@ -803,6 +804,56 @@ func TestCacheWraps(t *testing.T) {
 	require.IsType(t, cachemulti.Store{}, cacheWrappedWithListeners)
 }
 
+func TestTraceConcurrency(t *testing.T) {
+	db := dbm.NewMemDB()
+	multi := newMultiStoreWithMounts(db, types.PruneNothing)
+	err := multi.LoadLatestVersion()
+	require.NoError(t, err)
+
+	b := &bytes.Buffer{}
+	key := multi.keysByName["store1"]
+	tc := types.TraceContext(map[string]interface{}{"blockHeight": 64})
+
+	multi.SetTracer(b)
+	multi.SetTracingContext(tc)
+
+	cms := multi.CacheMultiStore()
+	store1 := cms.GetKVStore(key)
+	cw := store1.CacheWrapWithTrace(b, tc)
```

Comment:
How does the existence of this wrapper affect the outcome even though it's not used?

I'm trying to backport this fix for https://github.com/cosmos/cosmos-sdk/pull/10174 and want to make sure I fully understand the race condition.